src/bases/DataProviderBase.ts

```diff
@@ -0,0 +1,86 @@
+import { Observable, Observer } from '@dojo/core/Observable';
+import { DataProperties, SortDetails } from '../interfaces';
+
+export interface DataProviderConfiguration {
+	sort?: SortDetails | SortDetails[];
+}
+
+export interface Options {
+	[option: string]: any;
+	configuration?: DataProviderConfiguration;
+}
+
+export interface DataProviderState<O extends Options> {
+	options: O;
+	sort?: SortDetails[];
+}
+
+abstract class DataProviderBase<T, O extends Options> {
+	private data: DataProperties<T>;
+	private state: DataProviderState<O>;
+
+	observable: Observable<DataProperties<T>>;
```

Comment:
I am likely missing something, but why is this an external concern by anyone?  Especially since you have `.observe()` which returns this.